api/app.js

```diff
@@ -30,38 +35,37 @@ app.use(function(req, res, next) {
 
 
 // check whether the request has a valid JWT access token
-let authenticate = (req, res, next) => {
+let authenticate = async (req, res, next) => {
     let token = req.header('x-access-token');
 
     // verify the JWT
-    jwt.verify(token, User.getJWTSecret(), (err, decoded) => {
-        if (err) {
-            // there was an error
-            // jwt is invalid - * DO NOT AUTHENTICATE *
-            res.status(401).send(err);
-        } else {
-            // jwt is valid
-            req.user_id = decoded._id;
-            next();
-        }
-    });
+
+    try {
+        let decoded = await verify(token, User.getJWTSecret());
+        // jwt is valid
+        // eslint-disable-next-line require-atomic-updates
+        req.user_id = decoded._id;
+        next();
+    } catch (e) {
+        // there was an error
+        // jwt is invalid - * DO NOT AUTHENTICATE *
+        res.status(401).send(e);
+    }
 }
 
 // Verify Refresh Token Middleware (which will be verifying the session)
-let verifySession = (req, res, next) => {
+let verifySession = async (req, res, next) => {
     // grab the refresh token from the request header
     let refreshToken = req.header('x-refresh-token');
 
     // grab the _id from the request header
     let _id = req.header('_id');
 
-    User.findByIdAndToken(_id, refreshToken).then((user) => {
-        console.log(user);
+    try {
+        let user = await User.findByIdAndToken(_id, refreshToken)
         if (!user) {
             // user couldn't be found
-            return Promise.reject({
-                'error': 'User not found. Make sure that the refresh token and user id are correct'
-            });
+            throw new Error('User not found. Make sure that the refresh token and user id are correct')
```

Comment:
Since the function is now async, and async functions return promises, throwing an error is the exact same as rejecting the promise